src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -2011,48 +2037,42 @@ private String init(boolean initFull) {
                 lockedDueToDcmUpload = true;
             }
             //This is a hack to remove dataset locks for File PID registration if 
-                //the dataset is released
-                //in testing we had cases where datasets with 1000 files were remaining locked after being published successfully
-                /*if(dataset.getLatestVersion().isReleased() && dataset.isLockedFor(DatasetLock.Reason.pidRegister)){
-                    datasetService.removeDatasetLocks(dataset.getId(), DatasetLock.Reason.pidRegister);
-                }*/
-            if (dataset.isLockedFor(DatasetLock.Reason.pidRegister)) {
+            //the dataset is released
+            //in testing we had cases where datasets with 1000 files were remaining locked after being published successfully
+            /*if(dataset.getLatestVersion().isReleased() && dataset.isLockedFor(DatasetLock.Reason.finalizePublication)){
+                datasetService.removeDatasetLocks(dataset.getId(), DatasetLock.Reason.finalizePublication);
+            }*/
+            if (dataset.isLockedFor(DatasetLock.Reason.finalizePublication)) {
+                // "finalizePublication" lock is used to lock the dataset while 
+                // the FinalizeDatasetPublicationCommand is running asynchronously. 
+                // the tasks currently performed by the command are the  pid registration 
+                // for files and (or) physical file validation (either or both 
+                // of these two can be disabled via database settings). More 
+                // such asynchronous processing tasks may be added in the future. 
                 JH.addMessage(FacesMessage.SEVERITY_WARN, BundleUtil.getStringFromBundle("dataset.publish.workflow.message"),
                         BundleUtil.getStringFromBundle("dataset.pidRegister.workflow.inprogress"));
             }
+            if (dataset.isLockedFor(DatasetLock.Reason.FileValidationFailed)) {
+                // the dataset is locked, because one or more datafiles in it 
+                // failed validation during an attempt to publish it. 
+                if (FacesContext.getCurrentInstance().getExternalContext().getFlash().get("errorMsg") == null) {
+                    JH.addMessage(FacesMessage.SEVERITY_ERROR, BundleUtil.getStringFromBundle("dataset.publish.file.validation.error.message"),
+                            BundleUtil.getStringFromBundle("dataset.publish.file.validation.error.contactSupport"));
+                }
+                /* and now that we've shown the message to the user - remove the lock? */
```

Comment:
Because we want the admin to intercede here we probably don't want to remove the lock, but should we update the  edit data button so that if there is a file validation failure the user cannot add files? We would still allow them to make other edits?

Reply:
Or maybe use the file validation failure lock to disable the publish button?

Reply:
No, we don't want to remove that lock automatically - that was an idea I was entertaining during the development. I should remove that comment.
As for the second comment - that's exactly what that "failed validation" lock does. It disables the publish button, and will prevent an execution of the publish command, if called via API. But it shouldn't prevent the contributor from making changes. But yes, this should be documented clearly.

Reply:
OK. I missed that the lock disables publish because it's if it's locked for any reason, not just file validation failure.

Reply:
The way that page is dealing with/checking for various types of locks is a little scary.
Added a quick note to the "dataset management"/"publish dataset" section of the user guide.